x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/analysis/Analyzer.java

```diff
@@ -42,4 +60,110 @@ private LogicalPlan verify(LogicalPlan plan) {
         }
         return plan;
     }
+
+    private static class ReplaceWildcards extends AnalyzeRule<Filter> {
+
+        private static boolean isWildcard(Expression expr) {
+            if (expr instanceof Literal) {
+                Literal l = (Literal) expr;
+                if (l.value() instanceof String) {
+                    String s = (String) l.value();
+                    return s.contains("*");
+                }
+            }
+
+            return false;
+        }
+
+        private static LikePattern toLikePattern(String s) {
+            // pick a character that is guaranteed not to be in the string, because it isn't allowed to escape itself
+            char escape = 1;
+
+            // replace wildcards with % and escape special characters
+            String likeString = s.replace("%", escape + "%")
+                                 .replace("_", escape + "_")
+                                 .replace("*", "%");
+
+            return new LikePattern(likeString, escape);
+        }
+
+        @Override
+        protected LogicalPlan rule(Filter filter) {
+            return filter.transformExpressionsUp(e -> {
+                // expr == "wildcard*phrase"
+                if (e instanceof Equals) {
+                    Equals eq = (Equals) e;
+
+                    if (isWildcard(eq.right())) {
+                        String wcString = (String) ((Literal) eq.right()).value();
+                        e = new Like(e.source(), eq.left(), toLikePattern(wcString));
+                    }
+                }
+
+                // expr != "wildcard*phrase"
```

Comment:
I doubt that the functionality has been leveraged by many users, but if a wildcard string is on the left instead of the right, then it's just a normal, non-wildcard comparison and will check for equality or inequality depending on the operator.

Reply:
That seems a bit strange to imho. I'd find it more intuitive if `==` and `!=` would work in the same way regardless of right/left positioning of the args.

Reply:
>if a wildcard string is on the left instead of the right, then it's just a normal, non-wildcard comparison and will check for equality or inequality depending on the operator.

Which would make `==` and `!=` non commutative which is a bug.